packages/hix/lib/Hix/Options.hs

```diff
@@ -249,13 +254,25 @@ ghcidParser cwd = do
   extra <- extraGhcidParser
   pure GhcidOptions {..}
 
-newParser :: Parser NewOptions
-newParser = do
-  name <- strOption (long "name" <> short 'n' <> help "The name of the new project and its main package")
+initCommonParser :: Parser NewProjectConfigCommon
+initCommonParser = do
   packages <- switch (long "packages" <> short 'p' <> help "Store packages in the 'packages/' subdirectory")
   hixUrl <- strOption (long "hix-url" <> help "The URL to the Hix repository" <> value def)
   author <- strOption (long "author" <> short 'a' <> help "Your name" <> value "Author")
-  pure NewOptions {config = NewProjectConfig {..}}
+  pure NewProjectConfigCommon {..}
+
+initParser :: Parser InitOptions
+initParser = do
+  name <- strOption (long "name" <> short 'n' <> help "The name of the new project and its main package")
+  config <- initCommonParser
+  pure InitOptions {config = InitProjectConfig {..}}
+
+newParser :: Parser NewOptions
+newParser = do
+  directory <- strArgument (metavar "DIR" <> help "Directory to create for the project")
```

Comment:
```suggestion
  directory <- strArgument (metavar "DIR" <> help "Directory to create for the project, last component used as project name")
```

We could still allow an optional explicit `--name`, and then say "used as project name default"